### DISCUSSION TITLE
Battery Symbols

### DISCUSSION THREAD

**Discussion by wKovacs** [erikw/tmux-powerline#349] [Q&A] [answered]
Moin,

I like your project!

I noticed that the battery symbols are not displayed or just placeholders. Do I use a wrong font? How can I fix this?

Regards

wKovacs

**Comment by erikw** [comment, marked as answer]
Hey,

it could be related to a recent change to the icons: https://github.com/erikw/tmux-powerline/pull/346

* Have you properly installed and configured your terminal to use a font with powerline symbols? https://github.com/erikw/tmux-powerline#requirements
* Try checking out the commit before this PR and try: 6e6612b

**Comment by wKovacs** [comment]
Moin,

sry for this late of an answer but life came in the way... anyways

I checked [e2334b1] an there I was able to see the symbols. In a browser and my konsole. In all newer versions I only saw placeholder.

I used the unicode package trying to identify the symbols and for the newer symbols I got following message:

`unicode 󱊣
U+F12A3  - No such unicode character name in database
UTF-8: f3 b1 8a a3 UTF-16BE: db84dea3 Decimal: &#987811; Octal: \03611243
󱊣
Category: Co (Other, Private Use); East Asian width: A (ambiguous)
Unicode block: F0000..FFFFF; Supplementary Private Use Area-A
Bidi: L (Left-to-Right)`

while the older symbols result in:
`unicode ♥
U+2665 BLACK HEART SUIT
UTF-8: e2 99 a5 UTF-16BE: 2665 Decimal: &#9829; Octal: \023145
♥
Category: So (Symbol, Other); East Asian width: A (ambiguous)
Unicode block: 2600..26FF; Miscellaneous Symbols
Bidi: ON (Other Neutrals)`

So my guess would be the new symbols are not included in any of my powerline fonts

Regards

**Reply by xx4h** [reply]
Hi wKovacs,

`unicode` is not checking the actual font, but the unicode database for known entries.

To find out if you have a font installed who does support the symbol/icon, you can use `fc-list`, like so
```
fc-list :charset=F12A3
```

I guess many of us are using nerd-fonts for some time now, you can find them here: https://www.nerdfonts.com/

But i also will put the battery segment on my list to make the symbols/icons configurable through the user config as well.

**Reply by wKovacs** [reply]
Thank you!!

I learned something new today ;)
fc-list :charset=F12A3 comes back empty but fc-list :charset=2665 gives me a list of fonts

I installed some nerd fonts an now I can see battery icons instead a place holder.

best regards
